### PR TITLE
548 add an endpoint which delete an imported asset

### PR DESCRIPTION
**Pull request by B-Naoufal**

### CODE REVIEW

**Review by peterjah — APPROVED** [review]

File: pkg/assets/asset.go [hunk 1]
```diff
@@ -119,11 +119,8 @@ func (s *AssetsStore) AssetExists(walletNickname, contractAddress string) bool {
 	return assetFound
 }
 
-// AddAsset adds the asset information for a given wallet nickname in the JSON.
-func (s *AssetsStore) AddAsset(walletNickname, assetAddress string, assetInfo models.AssetInfo) error {
-	// Update the ContractAssets map with the new asset information
-	s.AddAssetToMemory(walletNickname, assetAddress, assetInfo)
-
+// synchronizeMemoryToJSON converts the AssetsStore map to JSON format and writes it to the file.
+func (s *AssetsStore) synchronizeMemoryToJSON() error {
```
Comment (Thykof) [inline pkg/assets/asset.go:123]: why not naming this function `persist`?

**Review by Thykof — CHANGES_REQUESTED** [review]

File: api/walletApi-V0.yml [hunk 2]
```diff
@@ -397,7 +397,7 @@ paths:
           type: string
           pattern: ^AS[0-9a-zA-Z]+$
           required: true
-          description: The asset address (MRC-20 token address) to add to the wallet. It must start with "AS" and contain only alphanumeric characters.
+          description: The asset address (token address) to add to the wallet. It must start with "AS" and contain only alphanumeric characters.
```
Comment (Thykof) [inline api/walletApi-V0.yml:400]:
```suggestion
          description: The asset address (token address) to add to the account. It must start with "AS" and contain only alphanumeric characters.
```

File: api/walletApi-V0.yml [hunk 3]
```diff
@@ -437,6 +437,38 @@ paths:
           description: Internal Server Error - The server has encountered a situation it does not know how to handle.
           schema:
             $ref: "#/definitions/Error"
+    delete:
+      operationId: DeleteAsset
+      description: Delete token information from a wallet.
```
Comment (Thykof) [inline api/walletApi-V0.yml:442]:
```suggestion
      description: Delete token information from an account.
```

File: pkg/assets/asset.go [hunk 4]
```diff
@@ -189,6 +199,37 @@ func (s *AssetsStore) AddAssetToMemory(walletNickname, assetAddress string, asse
 	s.Assets[walletNickname] = walletAssets
 }
 
+// DeleteAssetFromMemory removes the asset information for a given wallet nickname and asset address from the AssetsStore.
```
Comment (Thykof) [inline pkg/assets/asset.go:202]:
```suggestion
// DeleteAssetFromMemory removes the asset information for a given account nickname and asset address from the AssetsStore.
```

File: pkg/assets/asset.go [hunk 5]
```diff
@@ -189,6 +199,37 @@ func (s *AssetsStore) AddAssetToMemory(walletNickname, assetAddress string, asse
...
+func (s *AssetsStore) DeleteAssetFromMemory(walletNickname, assetAddress string) {
```
Comment (Thykof) [inline pkg/assets/asset.go:203]:
```suggestion
func (s *AssetsStore) DeleteAssetFromMemory(nickname, assetAddress string) {
```

File: pkg/assets/asset.go [hunk 6]
```diff
@@ -189,6 +199,37 @@ func (s *AssetsStore) AddAssetToMemory(walletNickname, assetAddress string, asse
...
+
+	// Delete the asset from the ContractAssets map of the specific *assets.AssetsStore
+	delete(walletAssets.ContractAssets, assetAddress)
+
+	// Update the asset information in the AssetsStore
+	s.Assets[walletNickname] = walletAssets
+}
+
+// DeleteAsset deletes the asset information for a given wallet nickname in the JSON.
+func (s *AssetsStore) DeleteAsset(walletNickname, assetAddress string) error {
```
Comment (Thykof) [inline pkg/assets/asset.go:222]:
```suggestion
func (s *AssetsStore) DeleteAsset(nickname, assetAddress string) error {
```

**Review by Thykof — CHANGES_REQUESTED** [review]

**Review by Thykof — APPROVED** [review]

**Pull request merged**